### ISSUE TITLE
Include the Open Contracting Data Standard (OCDS) among the data publication formats

### ISSUE THREAD

**Issue by karolisgranickas** [eForms/eForms#260]
While we recognize that the consultation does not address the data format, we would like to recommend that the Open Contracting Data Standard (OCDS) be among the data formats used for publication (access and notification), in order to further increase the use of this information. The Open Contracting Partnership would be happy to offer assistance, including in mapping eForms to OCDS and in developing technical solutions to publish TED data in OCDS format.

Below are the key reasons why we believe publishing in OCDS format would tremendously benefit user communities and the European Union at large: 

**1. OCDS is a globally recognized open data standard for public contracting information**
The Open Contracting Data Standard (OCDS) is the most globally recognized open data standard for collecting and publishing information on public contracting processes. It makes this important information shareable, reusable and machine-readable so that users around the world can perform analyses, draw insights and innovate. The data [tools](https://www.open-contracting.org/resources/open-contracting-tools-directory/), user guides, [standard documentation](http://standard.open-contracting.org/) and other guidance around OCDS are continuously improving and growing in number, which can make it much easier for user communities to engage with TED data if published in OCDS. In addition, having TED data in OCDS can increase opportunities for the open contracting global community to give feedback on and help improve the quality of TED data. 

**2. TED data would see greater use, though the existing ecosystem of OCDS tools**
There are around 40 government agencies worldwide implementing the OCDS and its adoption is growing rapidly. There are currently around fifty OCDS-based open source online [tools](https://airtable.com/shrzycSNYRcmV0WSZ/tblhHNGcDXuievZ74?blocks=hide) that can potentially be re-applied to TED datasets, if published in OCDS. The availability of this plug-and-play tool ecosystem can increase the frequency and depth of use of TED data.  

**3. OCDS publication would help implement the EU strategies around public procurement and single market**
Well-structured and high-quality data can improve procurement outcomes and meaningfully contribute to the objectives of the Procurement and Single Market strategies, including promotion of SME participation in procurement, EU trade, cross-border procurement, competition, better corruption control, and uptake and measurement of strategic procurement. The more specific demonstration of this potential can be found in our recently developed [guidance](https://docs.google.com/document/d/10uHezdNkrdHtbIl6Uh3JZo7BlGm5T1qlCntqYGSVFg4/edit#) for contract registers and OCDS. 

**4. OCDS publication does not require a shift in legislation, nor it is a technically complex task**
Including OCDS among the data formats used for TED publication does not require a change in legislation and is a technically manageable task. The OCP has prepared detailed [guidance](http://standard.open-contracting.org/profiles/eu/master/en/) on mapping OCDS to the present standard forms for procurement, and is happy to update the mapping once the new eForms are confirmed. Furthermore, the OCDS Helpdesk is ready to assist in the development of technical solutions to publish OCDS.

**Comment by lederberger** [comment]
Agree, important initiative!

**Comment by Nestulia** [comment]
Very needed! Already done in Ukraine. Helps a lot!

**Comment by Izdeba** [comment]
I am very supportive for this idea. As the representative of the institution which counters corruption it is important for us to access well structured data and built tools like corruption risks indicators. The OCDS has a full potential to help us to save taxpayers money!

**Comment by elenacalistru** [comment]
This is so needed  in Romania as well! Good initiative!

**Comment by oleksiiyurchenko** [comment]
Great initiative!
I would emphasize the mentioned contribution to cross-border trade. Including the OCDS in data standards for publication of notices will facilitate access to notices by non-EU countries (e.g. GPA parties) who had already implemented the OCDS. 
Relatively small efforts to provide a potentially huge impact.

**Comment by andreipetre182** [comment]
Good initiative. We need this in Romania.

**Comment by myroslav** [comment]
I hereby confirm that OCDS helped a lot in Ukraine and in Moldova. 

The reasons mentioned are very important for new eForms adoption, accessibility, and visibility.

**Comment by ddie** [comment]
Data quality, structure and comparability represents a major challenge to making public procurement in Europe more effective, transparent and accountable. Hence it is a great suggestion to include an explicit note on mandatory use of an open data standard like the OCDS for the publication of public procurement data and documents to the eForms survey.

**Comment by Nathalie-v** [comment]
Important and far-reaching initiative! OCDS enables interoperability of data across countries and makes it possible to reuse existing tools with relatively lower costs.

**Comment by augur-akm** [comment]
Great initiative! 
It helps to create data that easily can be transformed into valuable information.

**Comment by paulakeen** [comment]
I do agree with the initiative that TED should publish the eForms in different syntax-bindings, OCDS is just one, but then you have also UBL-XML, UBL-JASON and hopefully soon OWL-DL-Turtle/XML/etc. The mechanism to get there should not, in my opinion, map directly the eForms into the OCDS but use the eProcurement Ontology (ePO) being developed by the European Commission as the knowledge-representation, Data Element Dictionary and syntax-agnostic artefact for the mapping of concepts and terms between the different existing syntaxes and their underlying semantics. **That is one main role of ontologies, and was one important point of the rationale for the development of the ontology**.

By the way:

1. I understood that the Open Contracting Partnership (OCP) is aware and contributes to the development of the ePO as members of the ePO Working Group. Therefore your knowledge on OCDS, eForms and the Ontology is "the" leverage all of us are needing.

2. The ePO eNotification and eAccess phases are being aligned to eForms. Hence the mapping should be seamless, and could be implemented quite soon;

3. The ePO is being represented as very expressive UML diagrams, which not only helps understand the relationships between each entity but also can be exported as a standard (XMI), which in turn could be used as an intermediate artefact to wire the ontology data elements onto any syntax desired (OCDS, UBL, JSON, RDF-based, whatever): something we missed a lot in ePO (the graphic representation) when studying OCDS with the objective of reusing the OCDS data model. But, of course, there are other mechanisms and solutions to produce intermediate mapping artefacts (e.g. OASIS Genericode, or possibly one of the fancy tools already existing for OCDS that could be customised?).

Reactions will be welcomed.
KR

**Comment by veronicacretu** [comment]
Fully support this idea. Having Open Contracting Data Standard (OCDS) among the data formats, would be of a great value added to data overall. And in addition to this, if OCP continues to further develop its MEL framework, its potential is even higher. There are few organizations that care about learning from what works and what does not!  And OCP if one of them!

**Comment by jpmckinney** [comment]
@paulakeen Indeed – we anticipate that a mapping would be made between OCDS and ePO, which would then be used to calculate a direct mapping between OCDS and eForms if desired/useful.

We will also continue to contribute to the ePO Working Group, and are soon adding more capacity at OCP to engage more regularly with it.

As you note, the OCDS data model is implicit to its format/syntax and is not explicitly expressed using UML or otherwise; at this point, we consider it more expedient to map to the ePO diagrams, rather than build our own diagrams.

**Comment by woodbine** [comment]
I'd thoroughly back this initiative. The current use of e-Forms is predicated on creating a legal document repository and does not adequately capture the flow of information between the different parts of the procurement process. The absence of any option to include adequate organisational identifiers in e-Forms highlights this division in purpose.
The opportunity to link the data in tenders and contracts is a hallmark of OCDS and would greatly increase the value of the e-Forms output.

**Comment by grogorgregor** [comment]
Very important initiative. Fully suport that!

**Comment by joelgombin** [comment]
That would indeed be very useful, and would be a very strong message sent to Member States and all stakeholders that interoperability through OCDS is the way forward.

**Comment by francescopwi** [comment]
Publish TED in OCDS it would be very valuable for our project cause will allow us to have high quality data and reliable metrics to measure effectiveness of the public spending in our territory. Fully support that!

**Comment by Bosman18** [comment]
I consdider this initiative as extremely useful. I hope it would respond to challenge of a low competition on public procurement market in Poland (the highest percentage of tenders with only one bid in Europe). Referring our experience with tool agreggating TED and Polish Public Procurement Bulletin datasets - http://barometrryzyka.pl/ I am pretty sure that one reconciled and immutable data format like OCDS would contribute to higher number of their users, hence to more competitive market and more effective civic control too.

**Comment by romtal** [comment]
Great initiative in order to create more interoperability on public procurement data and have a real visibility on the entire value chain of public procurement through OCDS

**Comment by GavinHayman** [comment]
Machine-readable, user-friendly open data such as the OCDS will be of huge value to the frontline, service delivery parts of government and will also allow a host of innovations and applications to be built off it. [You can see a library of relatively easily adaptable analytical tools for OCDS here. ](https://airtable.com/shrzycSNYRcmV0WSZ/tblhHNGcDXuievZ74?blocks=hide  )

There is also huge value in adopting an well-documented, tested open data schema for developers and third-party 'infomediaries' selling value-added services to governments for analysis and insight into their data and will help support the European tech ecosystem as providing insights to save governments money and time and improving service delivery. Another value of the OCDS is that it has been designed to be inter-operable with budget, corporate and payment information further boosting analysis and insight. 

The Open Contracting Partnership has already published a detailed mapping to [old and new EU forms](http://standard.open-contracting.org/profiles/eu/master/en/) and to [WTO GPA reporting](http://standard.open-contracting.org/profiles/eu/master/en/).

A free global helpdesk provides support and so should minimise costs of adoption for member states.

**Comment by clarnimhairtin** [comment]
Agree. At Transparency International we have seen the power of the OCDS particularly in Ukraine but increasingly in other countries! Having information available in such a format unlocks so many opportunities for corruption fighters!

**Comment by anabrandusescu** [comment]
Agree, this is a great proposal.

**Comment by sandervdwaal** [comment]
OCDS is emerging as the global standard for making open contracting data available. If TED data would be available as OCDS from the source, this would unlock a host of OCDS tools to be usable with TED and make it possible to compare open contracting information from various sources much more easily.  We would certainly welcome this at Open Knowledge International!

**Comment by molliehanley** [comment]
An excellent initiative. When procurement information is made available as open, structured and machine-readable data via OCDS, countless stakeholders benefit from reduced friction and increased transparency. We welcome the EU adopting this approach.

**Comment by timgdavies** [comment]
Great to see all the support for this proposal, which I would also back. The real advantage I see here is that if national advocacy can secure publication of 'below threshold' data in OCDS format (e.g. from local authorities), and above-threshold data is available in OCDS through TED, then building up a much more complete picture of public procurement becomes easier.

**Comment by chrisalexsmith** [comment]
A really sensible and practical idea with a huge number of benefits. As countries around the world start to recognise the benefits of a data standard for public procurement and the huge potential for a quick leap to digital procurement, this is a once in a decade opportunity for Europe to show the way forward and hopefully even more will follow the path to digital open contracting.

**Comment by JachymHercher** [comment]
Hello everyone,

Thank you very much for the proposal and the engagement. We are very glad to see so much support for open contracting!

You raise very valid points, in particular on the advantages of reusing tools for the OCDS. We will discuss your proposal with the Publications Office of the EU (@annalisavacca) and get back to you with a more detailed answer.

For our discussion, we would like to clarify what exactly you mean by "OCDS being the data format used for publication". Essentially, we see three use-cases:
1. Being able to submit an OCDS-compliant file for publication on TED.
2. Being able to download TED open data in OCDS.
3. (Being able to download individual TED notices in OCDS off of the TED website.)

For all three use-cases, the main technical challenge seems to be the translation of TED data into OCDS and back. You offer assistance in this area and have already [done impressive work on it](http://standard.open-contracting.org/profiles/eu/master/en/), so it seems this challenge can be overcome. 

If that is the case, then any OCDS-based system could install a "converter" from OCDS to TED and easily send TED-compliant notices. On the other hand, TED (or any other authority) could use this converter to convert the TED open data into OCDS open data. From a practical point of view, does this meet your goals? 

Alternatively, do you believe it would be beneficial to have a more "official" arrangement? If yes, may we ask for more information on how exactly you would see that working (in particular from the point of view of the governance of the OCDS; responsibility for potential bugs, etc.)?

**Comment by myroslav** [comment]
To reuse OCDS tools only technical component is necessary (TED -> OCDS converter) and there are several initiatives that had already done that based on current XML TED notices, including [EU profile of OCDS](http://standard.open-contracting.org/profiles/eu/master/en/). They can step in and reveal the areas where TED notices would need more attention. Usually it is in the area, where important information in TED notices is not structured enough to be analysed. The reason of this "reduced" level of structure is in Directives that are not detailed enough.

Submission OCDS-compliant file for publication in TED has another aspect - OCDS sometmes requires more details than Directives (and eForms) require to capture. This aspect touches upon legal dependencies that can be overcome with new revision of Directives, but out of current scope and timeframe.

To summarize OCDS and TED are mappable to some extent, but sometimes mapping is lossy, i.e. it is impossible to convert from one format and then back without losses. And the reason is more legal than technical. [EU profile of OCDS](http://standard.open-contracting.org/profiles/eu/master/en/) solves the question of lossless TED->OCDS conversion. The question of OCDS->TED conversion is still open, and something like "extensions" to eForms would help advent the compatibility. I was considering submitting separate GitHub issue regarding this eForms extensibility, @JachymHercher should I?

**Comment by esimperl** [comment]
I very much agree with the aims and ambition of the proposal and the points that were put forward by the community. Data standards, and open standards in particular,  are extremely important. They can make and break projects and foster innovation by removing frictions and allowing developers and other stakeholders to focus on added value products and services rather than tedious data wrangling.  As an academic with a background in open and linked data, I have experienced first hand the impact a community-driven standard can have on innovation. OCDS is widely used and appreciated. It is versatile in terms of semantics and applications, and can be extended to a range of applications. We use it successfully in They Buy For You, a multi-million research grant funded through H2020, where we develop procurement analytics for the private and the public sector, piloted in three European countries. OCDS gave us starting point for our work, which includes TED notices. Including OCDS support in TED would be invaluable for us as it reduce the need for mappings, which are notoriously costly to define and messy, and introduce another layer of complexity (and errors) in already complex data pipelines.

**Comment by karolisgranickas** [comment]
Thank you @Jachym for your response and your questions. 

Our understanding is that use case 2. “Being able to download TED open data in OCDS” is the most easily achieved from the mapping work that has been done and the changes that have been proposed to the eForms. We would be happy to support the Publications Office to enable this conversion. Once the data is available in OCDS serialization, we would also be happy to start supporting the use of the data by European stakeholders to advance improved procurement outcomes. How soon would an EU publication of OCDS serialization be possible do you think?

We are also happy to see that you are interested in OCDS to TED conversion.  This is very aligned with the goal of seeing more EU Member States publishing public contracting information as open data at source. The Commission is already promoting the idea of contract registers with Member States that would include TED notices but also include additional information than what is mandated by the Directives.  

Member States publishing standardized open data will result in knowledge and technical capacity to publish data around small value procurement, too, providing a really valuable, unified and higher quality dataset to drive reforms and measure impacts of procurement reforms and help meet wider European Single Market objectives such as increasing competition, public integrity, and cross-border trade. 

We believe publishing OCDS at source would be useful for the following reasons:

1. It can encourage diligence in Member States around data creation, collection, and quality of data published, as they would then be mapping the schema and integrating it with their own systems which will encourage them to get their own data models right and encourage their own at-source reforms. As mentioned above, a clear, well-documented shared schema such as the OCDS – which is versatile in terms of semantics and applications – can prevent costly, messy mapping and reduce complexity (and errors) in already complex data pipelines. Transparency applied ex-ante, rather than post-hoc, can have a significantly improved impact on the behaviour of publishers. The detailed research by Monika Bauhr, Agnes Czibik, Mihaly Fazekas and Jenny de Fine Licht of the Digiwhist consortium of 3.5 million TED tenders showed that getting publishers to think about what they are publishing, when and how is more effective before rather than after the event (see [here](https://www.open-contracting.org/2017/12/06/greater-transparency-calls-tenders-save-europe-billions/)). Put simply, if a Member State thinks about data and applies the OCDS to its own datasets, the process of mapping and analysis can identify how to improve their data structures, collections, and interoperability and is going to engender reflection more on how to improve its data, especially as some basic tools could then give it a data quality dashboard in real time to improve its performance further. 

2. It can promote interoperability of data within governments, i.e. linking procurement data to budget, spending and beneficial ownership information (all of which need unique identifiers). OCDS already included planning and implementation of contracts, and having Member States think of direct publication would also encourage them to think about linking up this information. OCP, together with the Global Initiative for Fiscal Transparency (GIFT), has developed [extensions](https://extensions.open-contracting.org/en/extensions/budget_and_spend/master/) covering linkages between OCDS and budget & spending to support this kind of work. 

3. Anticipating that above-threshold publication might inspire publication of small-value procurement information in Member States, it can benefit the wider European business ecosystem by providing information in a global, free, well-documented schema that saves time in finding and using data, especially for cross-border procurement opportunities, but also for the new business creation. It can also be instrumental in building an SME-friendly single market, in line with the [Small Business Act](https://ec.europa.eu/growth/smes/business-friendly-environment/small-business-act_en) of the EU. 

4. It can encourage data interoperability across Member States, improving comparative analysis, etc. Interoperability is relevant in the context of the EU neighborhood policy, especially in the case of Moldova, Ukraine and Georgia – all of which already publish procurement data in OCDS. Easier access to procurement markets are in line with the association agreements with these countries as well as the general neighborhood policy.

5. It can encourage re-use of TED data by civil society and other public interest actors across the EU, in line with G20 Anti-Corruption Working Group Strategy. The demand for it has already been demonstrated by projects, such as Digiwhist and www.redflags.eu.

6. It can contribute to better procurement management and monitoring in Member States, in line with the EU Single Market and Public Procurement strategies. The OCP, together with the EBRD is currently scoping out the work in a few Member States (including Portugal and Slovakia) and is planning to develop showcase projects demonstrating how open procurement data can help achieve some or all of the above-mentioned objectives. So, if you were looking at conversion only, that could be neatly done on the TED website publishing data in OCDS format (alongside other formats). Ideally, you could then integrate a series of basic visualisations and tools using our extensive [archive of tools and OCDS applications](https://www.open-contracting.org/resources/open-contracting-tools-directory/), giving anyone interested considerable functionality to play with the data and do basic analysis. Building on our previous work, we are happy to assist of course. 

For the reasons above, we hope that the Publications Office can recommend OCDS to Member States, and promote its adoption through the financial assistance that the European Commission might invest to align the data infrastructures of Member States given the new eForms.

When more Member States are publishing OCDS, it will be important to ensure that their publication is aligned for OCDS -> TED conversion as you rightly pointed out above. We are happy to collaborate with the Commission on supporting that this is done correctly and open to exploring different options for more official collaboration.

**Comment by woodbine** [comment]
Jachym / All

I would encourage the Commission to think beyond a simple TED -> OCDS data
mapping and back again. There are some features in OCDS that are superior
to TED's feature set, particularly in the matter of handling updates and
enhancements to the data and in the monitoring the implementation of a
contract in TED. There is also the option to publish a sign contract
document in a meaningful way.

If the EU wishes to maintain a different data standard, it would be helpful
if they consider whether their standard can meet the features offered by
OCDS. Personally, I would advocate using OCDS and then extending it to meet
the specific needs of the EU, as this approach avoids having to re-think or
rework the six years of work that has already been done on OCDS. So adding
a feature that allows publishers to record VEATs and other document types
that are enshrined in EU law is moderately trivial.

I realise that this is an idealistic position that may be hard to achieve,
however, if this isn't possible, it would be useful if the EC would
consider how to extend TED to include the same features as OCDS.

I'm also working with Elena on the TheyBuyForYou project and I know that
the participants would be happy to share our experience of working with TED
data and OCDS.

Regards,

Ian

**Comment by chrisalexsmith** [comment]
We already have a great proof of concept with Public Contracts Scotland publishing their OJEU notices in OCDS https://www.publiccontractsscotland.gov.uk/NoticeDownload/Download.aspx 
![image](https://user-images.githubusercontent.com/10654017/55216178-f91f6d80-51f3-11e9-9b83-140a68168b6c.png)

**Comment by biz-portal** [comment]
Tremendous initiative! Hopefully, the OCDS will be fully implemented in Bulgaria in the foreseeable future! The cases of Ukraine and Moldova are a great source of good practice techniques!
